packages/server/lib/browsers/index.ts

```diff
@@ -155,13 +156,36 @@ export = {
     // TODO: normalizing opening and closing / exiting
     // so that there is a default for each browser but
     // enable the browser to configure the interface
-    instance.once('exit', () => {
+    instance.once('exit', async (code, signal) => {
       ctx.browser.setBrowserStatus('closed')
       // TODO: make this a required property
       if (!options.onBrowserClose) throw new Error('onBrowserClose did not exist in interactive mode')
 
+      const browserDisplayName = instance?.browser?.displayName || 'unknown'
+
       options.onBrowserClose()
+      browserLauncher.clearInstanceState()
       instance = null
+
+      // We are being very narrow on when to restart the browser here. The only case we can reliably test the 'SIGTRAP' signal.
+      // We want to avoid adding signals in here that may intentionally be caused by a user.
+      // For example exiting firefox through either force quitting or quitting via cypress will fire a 'SIGTERM' event which
```

Comment:
> quitting via cypress will fire a 'SIGTERM' event which

Is this quitting like a hard kill app quit? I notice an errors randomly this when we click the close button or ctrl + c in the terminal for Cypress unexpectedly killing

Reply:
Yeah sigterm is what you get when you hard kill the app. Specifically in this case we hard kill firefox. But we can't use the sigterm signal because when we force exit firefox via the app, we get cause the sigterm signal and you end up in a loop where you can never kill firefox because we restart it.